Modules/Common/CMakeLists.txt

```diff
@@ -1,55 +1,43 @@
-set(MODULE_NAME "QcCommon")
-
-# ---- Files ----
-
-set(SRCS src/NonEmpty.cxx src/MeanIsAbove.cxx)
-
-set(HEADERS include/Common/NonEmpty.h include/Common/MeanIsAbove.h)
-
-# Produce the final Version.h using template Version.h.in and substituting variables. We don't want to polute our source
-# tree with it, thus putting it in binary tree.
-configure_file("include/Common/Version.h.in" "${CMAKE_CURRENT_BINARY_DIR}/include/Common/Version.h" @ONLY)
+# Produce the final Version.h using template Version.h.in and substituting
+# variables. We don't want to polute our source tree with it, thus putting it in
+# binary tree.
+configure_file("include/Common/Version.h.in"
+               "${CMAKE_CURRENT_BINARY_DIR}/include/Common/Version.h" @ONLY)
 
 # ---- Library ----
 
-add_library(${MODULE_NAME} SHARED ${SRCS} ${MODULE_NAME}Dict.cxx)
+add_library(QcCommon src/NonEmpty.cxx src/MeanIsAbove.cxx)
 
 target_include_directories(
-  ${MODULE_NAME}
```

Comment:
Unimportant : I would propose to keep this variable to avoid repeating ourselves many times. It is easier for the code generation of new modules based on the skeleton.

Reply:
I would argue that using the string QcSkeleton achieves the same thing, without relying on a custom cmake variable